diff -upNr AusweisApp2-1.18.2.orig/cmake/Install.cmake AusweisApp2-1.18.2.patched/cmake/Install.cmake
--- AusweisApp2-1.18.2.orig/cmake/Install.cmake	2019-09-30 17:25:27.000000000 +0200
+++ AusweisApp2-1.18.2.patched/cmake/Install.cmake	2019-11-20 19:14:16.464539619 +0100
@@ -3,7 +3,7 @@
 # It will be included by ./src/CMakeLists.txt
 ################################################
 
-SET(TRANSLATION_DESTINATION translations)
+SET(TRANSLATION_DESTINATION share/qt5/translations)
 SET(DEFAULT_FILE_DESTINATION .)
 
 IF(CMAKE_PREFIX_PATH)
@@ -78,7 +78,7 @@ IF(WIN32)
 		${DEPENDENCY_CHECK}
 		${SEARCH_ADDITIONAL_DIRS}
 		INCLUDE(BundleUtilities)
-		FIXUP_BUNDLE(\"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${PROJECT_NAME}${CMAKE_EXECUTABLE_SUFFIX}\" \"${libQuickControls2}\" \"${TOOLCHAIN_BIN_PATH};\${ADDITIONAL_DIRS}\")
+		#FIXUP_BUNDLE(\"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${PROJECT_NAME}${CMAKE_EXECUTABLE_SUFFIX}\" \"${libQuickControls2}\" \"${TOOLCHAIN_BIN_PATH};\${ADDITIONAL_DIRS}\")
 		" COMPONENT Runtime)
 
 
@@ -139,7 +139,7 @@ ELSEIF(APPLE AND NOT IOS)
 		file(GLOB_RECURSE QTPLUGINS \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${MACOS_BUNDLE_PLUGINS_DIR}/*${CMAKE_SHARED_LIBRARY_SUFFIX}\")
 		file(GLOB_RECURSE QtQuickPLUGINS \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${MACOS_BUNDLE_RESOURCES_DIR}/*${CMAKE_SHARED_LIBRARY_SUFFIX}\")
 		INCLUDE(BundleUtilities)
-		FIXUP_BUNDLE(\"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${MACOS_BUNDLE_MACOS_DIR}/${PROJECT_NAME}\" \"\${QTPLUGINS};\${QtQuickPLUGINS}\" \"${TOOLCHAIN_LIB_PATH};\${ADDITIONAL_DIRS}\")
+		#FIXUP_BUNDLE(\"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${MACOS_BUNDLE_MACOS_DIR}/${PROJECT_NAME}\" \"\${QTPLUGINS};\${QtQuickPLUGINS}\" \"${TOOLCHAIN_LIB_PATH};\${ADDITIONAL_DIRS}\")
 		" COMPONENT Runtime)
 
 	LIST(APPEND ADDITIONAL_BUNDLE_FILES_TO_SIGN "/Contents/Library/LoginItems/AusweisApp2AutostartHelper.app/Contents/MacOS/AusweisApp2AutostartHelper")
@@ -275,7 +275,7 @@ ELSEIF(UNIX)
 		${DEPENDENCY_CHECK}
 		${SEARCH_ADDITIONAL_DIRS}
 		INCLUDE(BundleUtilities)
-		FIXUP_BUNDLE(\"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${DEFAULT_FILE_DESTINATION}/${PROJECT_NAME}\" \"\" \"\${ADDITIONAL_DIRS}\")
+		#FIXUP_BUNDLE(\"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${DEFAULT_FILE_DESTINATION}/${PROJECT_NAME}\" \"\" \"\${ADDITIONAL_DIRS}\")
 		" COMPONENT Runtime)
 
 	CONFIGURE_FILE(${PACKAGING_DIR}/linux/AusweisApp2.desktop.in ${CMAKE_CURRENT_BINARY_DIR}/AusweisApp2.desktop @ONLY)
@@ -322,17 +322,26 @@ ELSEIF(NOT ANDROID_BUILD_AAR)
 	INSTALL(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/translations/ DESTINATION ${TRANSLATION_DESTINATION} COMPONENT Translations)
 ENDIF()
 
-IF(NOT ANDROID_BUILD_AAR)
+IF(NOT ANDROID_BUILD_AAR AND NOT LINUX)
 	# resources file
 	INSTALL(FILES ${RCC} DESTINATION ${DEFAULT_FILE_DESTINATION} COMPONENT Runtime)
 ENDIF()
 
-# qt qml plugins
-IF(DESKTOP AND TARGET Qt5::Qml)
-	FOREACH(entry QtQuick QtQuick.2 QtQml QtGraphicalEffects Qt)
-		INSTALL(DIRECTORY ${QT_HOST_PREFIX}/qml/${entry} DESTINATION ${DEFAULT_FILE_DESTINATION}/qml COMPONENT Runtime PATTERN "*.dylib" EXCLUDE)
-	ENDFOREACH()
+IF (LINUX)
+   INSTALL(FILES ${RCC} DESTINATION share/AusweisApp2 COMPONENT Runtime)
 ENDIF()
 
+# qt qml plugins
+#IF(DESKTOP AND TARGET Qt5::Qml)
+#	FOREACH(entry QtQuick QtQuick.2 QtQml QtGraphicalEffects Qt)
+#		INSTALL(DIRECTORY ${QT_HOST_PREFIX}/qml/${entry} DESTINATION ${DEFAULT_FILE_DESTINATION}/qml COMPONENT Runtime PATTERN "*.dylib" EXCLUDE)
+#	ENDFOREACH()
+#ENDIF()
+
 # secure storage file
-INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/config.json DESTINATION ${DEFAULT_FILE_DESTINATION} COMPONENT Runtime)
+
+IF (LINUX)
+   INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/config.json DESTINATION share/AusweisApp2 COMPONENT Runtime)
+ENDIF()
+
+
diff -upNr AusweisApp2-1.18.2.orig/cmake/Libraries.cmake AusweisApp2-1.18.2.patched/cmake/Libraries.cmake
--- AusweisApp2-1.18.2.orig/cmake/Libraries.cmake	2019-09-30 17:25:27.000000000 +0200
+++ AusweisApp2-1.18.2.patched/cmake/Libraries.cmake	2019-11-20 16:47:39.045751651 +0100
@@ -37,11 +37,13 @@ IF(ANDROID)
 ENDIF()
 
 SET(QT_HOST_PREFIX ${_qt5Core_install_prefix})
-FOREACH(dest "" "share/qt" "share/qt5")
-	IF(EXISTS "${QT_HOST_PREFIX}/${dest}/translations")
-		SET(QT_TRANSLATIONS_DIR ${QT_HOST_PREFIX}/${dest}/translations)
-	ENDIF()
-ENDFOREACH()
+#FOREACH(dest "" "share/qt" "share/qt5")
+#	IF(EXISTS "${QT_HOST_PREFIX}/${dest}/translations")
+#		SET(QT_TRANSLATIONS_DIR ${QT_HOST_PREFIX}/${dest}/translations)
+#	ENDIF()
+#ENDFOREACH()
+SET(QT_TRANSLATIONS_DIR QT_HOST_PREFIX}/share/qt5/translations)
+
 
 IF(NOT QT_TRANSLATIONS_DIR)
 	QUERY_QMAKE(QT_TRANSLATIONS_DIR QT_INSTALL_TRANSLATIONS)
diff -upNr AusweisApp2-1.18.2.orig/src/global/FileDestination.h AusweisApp2-1.18.2.patched/src/global/FileDestination.h
--- AusweisApp2-1.18.2.orig/src/global/FileDestination.h	2019-09-30 17:25:27.000000000 +0200
+++ AusweisApp2-1.18.2.patched/src/global/FileDestination.h	2019-11-20 16:50:29.593973453 +0100
@@ -41,6 +41,23 @@ class FileDestination
 		}
 
 
+		static QString getPrefixPath()
+		{
+			QString prefixPath;
+
+			const auto libPathes = QCoreApplication::libraryPaths();
+			for (const auto& path : libPathes)
+			{
+				prefixPath = path;
+				prefixPath.remove(QStringLiteral("/bin"));
+
+				if (prefixPath != path) {
+					break;
+				}
+			}
+
+			return prefixPath;
+		}
 };
 
 } // namespace governikus
diff -upNr AusweisApp2-1.18.2.orig/src/global/LanguageLoader.cpp AusweisApp2-1.18.2.patched/src/global/LanguageLoader.cpp
--- AusweisApp2-1.18.2.orig/src/global/LanguageLoader.cpp	2019-09-30 17:25:27.000000000 +0200
+++ AusweisApp2-1.18.2.patched/src/global/LanguageLoader.cpp	2019-11-20 16:50:29.593973453 +0100
@@ -23,7 +23,7 @@ const QLocale::Language LanguageLoader::
 QLocale LanguageLoader::mDefaultLanguage = QLocale::system();
 
 LanguageLoader::LanguageLoader()
-	: mPath(FileDestination::getPath(QStringLiteral("translations")))
+	: mPath(QStringLiteral("share/qt5/translations"))
 	, mTranslatorList()
 	, mComponentList(
 		{
@@ -31,6 +31,7 @@ LanguageLoader::LanguageLoader()
 		})
 	, mUsedLocale(mFallbackLanguage)
 {
+	mPath = FileDestination::getPrefixPath() + QStringLiteral("/") + mPath;
 }
 
 
diff -upNr AusweisApp2-1.18.2.orig/src/global/ResourceLoader.cpp AusweisApp2-1.18.2.patched/src/global/ResourceLoader.cpp
--- AusweisApp2-1.18.2.orig/src/global/ResourceLoader.cpp	2019-09-30 17:25:27.000000000 +0200
+++ AusweisApp2-1.18.2.patched/src/global/ResourceLoader.cpp	2019-11-20 16:50:29.597973460 +0100
@@ -42,7 +42,7 @@ void ResourceLoader::init()
 {
 	for (const auto& file : mFilenames)
 	{
-		QString path = FileDestination::getPath(file);
+		QString path = FileDestination::getPrefixPath() + QStringLiteral("/share/AusweisApp2/") + file;
 
 #ifndef QT_NO_DEBUG
 		const auto& filePathFromTestDir = QCoreApplication::applicationDirPath() + QStringLiteral("/../../src/") + file;
diff -upNr AusweisApp2-1.18.2.orig/src/secure_storage/SecureStorage.cpp AusweisApp2-1.18.2.patched/src/secure_storage/SecureStorage.cpp
--- AusweisApp2-1.18.2.orig/src/secure_storage/SecureStorage.cpp	2019-09-30 17:25:27.000000000 +0200
+++ AusweisApp2-1.18.2.patched/src/secure_storage/SecureStorage.cpp	2019-11-20 16:50:29.597973460 +0100
@@ -95,7 +95,8 @@ SecureStorage& SecureStorage::getInstanc
 
 void SecureStorage::load()
 {
-	const auto& path = FileDestination::getPath(QStringLiteral("config.json"));
+	QString prefixPath = FileDestination::getPrefixPath();
+	const auto& path = prefixPath + QStringLiteral("/share/AusweisApp2/config.json");
 
 	if (!QFile::exists(path))
 	{
